### PR TITLE
Crossdimensional Teleposition (including entities, into unloaded chunks etc)

### PR DESCRIPTION
**Pull request by keraldi**
Using code snippet from BrandonsCore (TeleportUtils), License permits this under line 22: using in other projects is allowed. I'll leave small credits in form of a comment above and below the code snipped. Removing everything stating permission.

Sidenote: Excused to brandon3055 for bothering him with an unneeded license query.

### CODE REVIEW

**Comment by TehNut** [comment]
You realize his license permits this already, right?

**Comment by keraldi** [comment]
Finished. Please review. Performance improvements might be possible.

File: src/main/java/WayofTime/bloodmagic/ritual/portal/Teleports.java [hunk 1]
```diff
@@ -115,28 +124,70 @@ public void teleport() {
...
+
                         if (!player.getEntityWorld().isRemote) {
                             SoulNetwork network = NetworkHelper.getSoulNetwork(networkOwner);
                             if (network.getCurrentEssence() < getTeleportCost())
                                 return;
 
-                            if (teleposer)
-                                if (MinecraftForge.EVENT_BUS.post(new TeleposeEvent.Ent(entity, entity.getEntityWorld(), entity.getPosition(), newWorldServer, new BlockPos(x, y, z))))
+                            if (teleposer) {
+                                if (MinecraftForge.EVENT_BUS.post(new TeleposeEvent.Ent(entity, entity.getEntityWorld(), entity.getPosition(), newWorldServer, targetTeleposer))) {
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/ritual/portal/Teleports.java]:
```java
if (teleposer && MinecraftForge.EVENT_BUS.post(new TeleposeEvent.Ent(entity, entity.getEntityWorld(), entity.getPosition(), newWorldServer, targetTeleposer))
```

File: src/main/java/WayofTime/bloodmagic/ritual/portal/Teleports.java [hunk 2]
```diff
@@ -115,28 +124,70 @@ public void teleport() {
...
+
+                            for (PotionEffect potioneffect : player.getActivePotionEffects()) {
+                                player.connection.sendPacket(new SPacketEntityEffect(player.getEntityId(), potioneffect));
+                            }
+                            FMLCommonHandler.instance().firePlayerChangedDimensionEvent(player, entity.dimension, newWorldID);
+                            player.setLocationAndAngles(x + 0.5, y + 0.5, z + 0.5, player.rotationYaw, player.rotationPitch);
+
+                            /* end brandon3055 teleportation code */
+
+                            ForgeChunkManager.releaseTicket(chunkTicket);
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/ritual/portal/Teleports.java]: The ticket is only being released when a player is being teleported. Any other mob will retain the ticket.

**Review by TehNut — CHANGES_REQUESTED** [review]

**Comment by keraldi** [comment]
Done.

**Review by TehNut — APPROVED** [review]

**Comment by MentalMouse** [comment]
Doesn't the teleposer swap source and destination?  That might be awkward with unloaded chunks.

**Comment by TehNut** [comment]
This loads the chunk if it's unloaded

**Pull request merged**